client/components/email-verification/email-verification-dialog/index.jsx

```diff
@@ -0,0 +1,118 @@
+import {
+	verifyEmail,
+	resetVerifyEmailState,
+} from 'state/current-user/email-verification/actions';
+
+class VerifyEmailDialog extends Component {
+
+	getResendButtonLabel() {
+		if ( 'sent' === this.props.emailVerification ||
+			'error' === this.props.emailVerification ) {
+			return this.props.translate( 'Email Sent' );
+		}
+		if ( 'requesting' === this.props.emailVerification ) {
+			return <Spinner className="email-verification-dialog__confirmation-dialog-spinner" />;
+		}
+		return this.props.translate( 'Resend Email' );
+	}
+
+	handleClose = () => {
+		this.props.resetVerifyEmailState();
+		this.props.onClose();
+	};
+
+	getDialogButtons() {
+		return [
+			<Button
+				key="resend"
+				primary={ false }
+				disabled={ includes( [ 'requesting', 'sent', 'error' ], this.props.emailVerification ) }
+				onClick={ this.props.verifyEmail }>
+				{ this.getResendButtonLabel() }
+			</Button>,
+			<Button
+				key="close"
+				primary={ true }
+				onClick={ this.handleClose }>
+					{ this.props.translate( 'OK' ) }
+			</Button>
+		];
+	}
+
+	render() {
+		const strings = {
+			confirmHeading: this.props.translate( 'Please confirm your email address' ),
+
+			confirmExplanation: this.props.translate( 'We sent you an email when you first signed up. ' +
+				'Please open the message and click the blue button.' ),
+
+			confirmReasoning: this.props.translate( 'Email confirmation allows us to assist when recovering ' +
+				'your account in the event you forget your password.' ),
+
+			confirmEmail: this.props.translate(
+				'{{wrapper}}%(email)s{{/wrapper}} {{emailPreferences}}change{{/emailPreferences}}',
+				{
+					components: {
+						wrapper: <span className="email-verification-dialog__confirmation-dialog-email-wrapper" />,
+						emailPreferences: <a href="/me/account" />
+					},
+					args: {
+						email: this.props.email
+					}
+				}
+			)
+		};
+
+		return (
+			<Dialog
+				isVisible={ true }
+				buttons={ this.getDialogButtons() }
+				additionalClassNames="email-verification-dialog__confirmation-dialog is-narrow"
+			>
+				<h1 className="email-verification-dialog__confirmation-dialog-heading is-variable-height">{ strings.confirmHeading }</h1>
+				<p className="email-verification-dialog__confirmation-dialog-email">{ strings.confirmEmail }</p>
+				<p className="email-verification-dialog__confirmation-dialog-explanation">{ strings.confirmExplanation }</p>
+				<p className="email-verification-dialog__confirmation-dialog-reasoning">{ strings.confirmReasoning }</p>
+			</Dialog>
+		);
+	}
+}
+
+VerifyEmailDialog.propTypes = {
+	onClose: React.PropTypes.func,
+	translate: React.PropTypes.func,
+	// connected props:
+	email: React.PropTypes.string,
+	emailVerification: React.PropTypes.string,
+};
+
+VerifyEmailDialog.defaultProps = {
+	onClose: noop
+};
+
+export default connect(
+	state => ( {
+		email: getCurrentUserEmail( state ),
+		emailVerification: get( state, 'currentUser.emailVerification.status' ),
```

Comment:
Maybe `emailVerificationStatus` as the prop name, so we have a little bit more context.